sigopt/objects.py

```diff
@@ -501,20 +501,54 @@ class Project(ApiObject):
   metadata = Field(Metadata)
 
 
+class Datasets(_DictWrapper):
+  pass
```

Comment:
The json representation of the training run's datasets is like this:
```json
{
  "datasets": {
    "mnist": {"object": "dataset"},
    "cifar10": {"object": "dataset"},
  }
}
```

This makes datasets extensible (we can add new fields to a dataset in the future) but I think there's some flexibility on the client side with how we present these to users.

Another option is to use a `set` or `list` interface instead of a `dict`, although your implementation is almost identical in functionality (ex. `"mnist" in run.datasets` and `for dataset_name in run.datasets`).